src/extras/transformers/fly-transformer.js

```diff
@@ -0,0 +1,205 @@
+import { Vec2 } from '../../core/math/vec2.js';
+import { Vec3 } from '../../core/math/vec3.js';
+import { Quat } from '../../core/math/quat.js';
+import { math } from '../../core/math/math.js';
+import { Transformer } from './transformer.js';
+
+/** @import { Mat4 } from '../../core/math/mat4.js'; */
+
+/**
+ * @typedef {object} FlyInputFrame
+ * @property {Vec3} move - The move deltas.
+ * @property {Vec2} rotate - The rotate deltas.
+ */
+
+const tmpV1 = new Vec3();
+const tmpV2 = new Vec3();
+const tmpQ1 = new Quat();
+
+/**
+ * Calculate the lerp rate.
+ *
+ * @param {number} damping - The damping.
+ * @param {number} dt - The delta time.
+ * @returns {number} - The lerp rate.
+ */
+const lerpRate = (damping, dt) => 1 - Math.pow(damping, dt * 1000);
+
+/**
+ * The fly controller.
+ *
+ * @category Transformers
```

Comment:
I'm thinking it might make more sense to group `InputDevice` and `Transformers`  classes together. Maybe under `Camera Controls`?